frontend/src/routes/_dashboard/index.lazy.tsx

```diff
@@ -19,8 +60,181 @@ function AuthedView() {
+                          ? "border-yellow-600 bg-yellow-600 bg-opacity-10 text-statusyellow"
+                          : ""
+                  }`}
+                >
+                  <div
+                    className={`border w-2 h-2 text-white  border-transparent p-0.5 rounded-full ${
+                      project.tracker === 1
+                        ? "bg-green-600"
+                        : project.tracker === 0
+                          ? "bg-red-600"
+                          : project.tracker === 2
+                            ? "bg-yellow-600"
+                            : ""
+                    }`}
+                  ></div>
+                  {project.status}
+                </div>
+              </TableCell>
+              <TableCell className="flex justify-end">
+                <div className="w-fit flex items-center gap-3">
+                  Settings
+                  <Settings width={18} />
+                </div>
+              </TableCell>
+            </TableRow>
+          ))}
+        </TableBody>
+      </Table>
+
+      <div className="my-4">
+        <DatatablePagination />
+      </div>
+    </main>
+  );
+}
+
+function DatatablePagination() {
+  return (
+    <div className="flex  items-center justify-end px-2">
+      <div className="flex items-center space-x-2">
+        <p className="text-sm font-medium">Rows per page</p>
+        <Select value="10">
+          <SelectTrigger className="h-8 w-[70px]">
+            <SelectValue placeholder="10" />
+          </SelectTrigger>
+          <SelectContent className="border border-border" side="top">
+            {[10, 20, 30, 40, 50].map((pageSize) => (
+              <SelectItem key={pageSize} value={`${pageSize}`}>
+                {pageSize}
+              </SelectItem>
+            ))}
+          </SelectContent>
+        </Select>
+      </div>
+
+      <div className="flex  items-center space-x-6 lg:space-x-8">
+        <div className="flex w-[100px] items-center justify-center text-sm font-medium">
+          Page of 10
+        </div>
+        <div className="flex items-center space-x-2">
+          <Button variant="outline" className="hidden h-8 w-8 p-0 lg:flex">
+            <span className="sr-only">Go to first page</span>
+            <ChevronsLeft className="h-4 w-4" />
+          </Button>
+          <Button variant="outline" className="h-8 w-8 p-0">
+            <span className="sr-only">Go to previous page</span>
+            <ChevronLeft className="h-4 w-4" />
+          </Button>
+          <Button variant="outline" className="h-8 w-8 p-0">
+            <span className="sr-only">Go to next page</span>
+            <ChevronRight className="h-4 w-4" />
+          </Button>
+          <Button variant="outline" className="hidden h-8 w-8 p-0 lg:flex">
+            <span className="sr-only">Go to last page</span>
+            <ChevronsRight className="h-4 w-4" />
+          </Button>
+        </div>
+      </div>
+    </div>
```

Comment:
This should be its own component, and named `Pagination` simply with these props : 


```tsx
export type PaginationProps = {
  totalPages: number;
  currentPage: number;
  perPage: number;
  className?: string;
  onChangePage: (page: number) => void;
  onChangePerPage: (perPage: number) => void;
};
```

Depending on the `perPage`, it should update the select value, and when the select value changes, it should trigger `onChangePerPage`.

The `currentPage` and `totalPages` should be used for the text displayed `"Page x out of Y"`. 

`currentPage` and `totalPages` need also to be used for the arrows buttons, the `previous` button should trigger `onChangePage(currentPage - 1)`, the `next` button triggers  `onChangePage(currentPage + 1)`, the `first` button triggers `onChangePage(1)` and the `last` button triggers  `onChangePage(totalPages)`. 

When the `currentPage` is `1` (the first page), the `"first"` and `"previous"`  button arrow should be disabled because you can't go back, this is the same for the `"last"` and `"next"` buttons when the `currentPage` is equal to `totalPages`. 


You can also inspire yourself from this component : https://github.com/Fredkiss3/gh-next/blob/main/src/components/pagination.tsx

Reply:
The state should be controlled by the parent of the component who calls `Pagination`